include/ifc/index-utils.hxx

```diff
@@ -173,7 +177,9 @@ namespace index_like {
     struct Over {
         using SortType = S;
         constexpr Over() : tag(), value() {}
-        constexpr Over(S s, uint32_t v) : tag(ifc::to_underlying(s)), value(v) {}
+        constexpr Over(S s, uint32_t v)
+          : tag(ifc::to_underlying(s) & bit_field_mask<tag_precision<S>>), value(v & bit_field_mask<index_precision<S>>)
+        {}
```

Comment:
Could you expand on here in the review comment why this is needed?

Reply:
This seems to be the only way to squelch the `-Wconversion` warning. I find it silly too, but the alternative solution would involve macros for GCC and Clang to disable the warning for the ctor.

Reply:
Ah, thanks!  In this specific case, I suggest to keep the original code in face of this warning.